### PR TITLE
fix: support mtp for qwen3-next

### PR DESCRIPTION
**Pull request by huang3eng**
This pull request adds support for MTP (Multi-Task Prediction) layers to the Qwen3 Next model integration for issue #1496.  It introduces logic for handling MTP-specific parameters and weight conversions during model bridging and conversion, ensuring compatibility between the Megatron and Hugging Face formats. The main changes are grouped into support for MTP layers, parameter/weight conversion logic, and configuration handling.

**Support for MTP layers:**

* Added `--mtp-num-layers` argument to the Qwen3 Next model shell script to allow specifying the number of MTP layers.
* Integrated MTP block specification into the Qwen3 Next bridge by overriding `_get_gptmodel_args` to include `mtp_block_spec` when MTP layers are configured.

**Parameter and weight conversion logic:**

* Implemented MTP-specific logic in `convert_qwen3_next_to_hf` to map MTP layer parameters from Megatron to Hugging Face format, including custom handling for special weights and recursive mapping for transformer sublayers.
* Overrode `_weight_name_mapping_mcore_to_hf` and `_convert_mtp_param` in the Qwen3 Next bridge to handle MTP parameter name conversions, including direct mappings and recursive handling for transformer subcomponents.
* Added logic in `_weight_to_mcore_format` and `_weight_to_hf_format` to handle special tensor permutations for `eh_proj.weight` parameters in both conversion directions.

**Configuration handling:**

* Extended `_build_config` to propagate the number of MTP layers from the Hugging Face config to the Megatron config, ensuring consistent setup. [[1]](diffhunk://#diff-83714a4a283afd385946897fbafe809f106b3a97724b4c6cd1a637f1eea48f7cL76-R152) [[2]](diffhunk://#diff-83714a4a283afd385946897fbafe809f106b3a97724b4c6cd1a637f1eea48f7cR175)
* Imported the required MTP block spec utility function in the Qwen3 Next bridge.

### Test
1. test qwen3-next hf to torch_dict
[hf_to_torch_dist_success.log](https://github.com/user-attachments/files/24883581/hf_to_torch_dist_success.log)
2. test qwen3-next torch_dict to hf
[torch_dist_to_hf_success.log](https://github.com/user-attachments/files/24883584/torch_dist_to_hf_success.log)
3. original hf model and converted hf model accept length basically maintain consistency
```
python3 -m sglang.launch_server --model /root/Qwen3-Next-80B-A3B-Thinking/ --tp 4 --speculative-num-steps 3  --speculative-eagle-topk 1  --speculative-num-draft-tokens 4 --speculative-algo NEXTN

python -m sglang.bench_one_batch_server --base-url http://127.0.0.1:30000 --model-path /root/Qwen3-Next-80B-A3B-Thinking/  --batch-size 1024  --input-len 8192 --output-len 512

python3 -m sglang.launch_server --model /root/Qwen3-Next-80B-A3B-Thinking-test/ --tp 4 --speculative-num-steps 3  --speculative-eagle-topk 1  --speculative-num-draft-tokens 4 --speculative-algo NEXTN

python -m sglang.bench_one_batch_server --base-url http://127.0.0.1:30000 --model-path /root/Qwen3-Next-80B-A3B-Thinking-test/  --batch-size 1024  --input-len 8192 --output-len 512

```
4. During the sft and rl training process, the mtp loss converges normally.

### CODE REVIEW

File: slime/backends/megatron_utils/megatron_to_hf/qwen3_next.py [hunk 1]
```diff
@@ -1,9 +1,55 @@
 import re
-
 import torch
 
-
 def convert_qwen3_next_to_hf(args, name, param):
+    # ================== MTP Logic Start ==================
+    if "mtp.layers" in name:
+        # 解析层号
+        # name 格式类似于 module.module.mtp.layers.0.enorm.weight
```
Comment (Zhuohao-Li) [inline slime/backends/megatron_utils/megatron_to_hf/qwen3_next.py]: please use English as the main language, rm unnecessary comments, and make the code structured

File: scripts/models/qwen3-next-80B-A3B.sh [hunk 2]
```diff
@@ -55,4 +55,7 @@ MODEL_ARGS=(
    # qwen3 specific
    --attention-output-gate
    --moe-shared-expert-gate
+
+   # mtp
+   --mtp-num-layers 1
```
Comment (Zhuohao-Li) [inline scripts/models/qwen3-next-80B-A3B.sh]: `--mtp-num-layers 1` should not be hardcoded in models/script, it is better to pass args in training script when start the job (https://github.com/THUDM/slime/blob/main/scripts/run-mimo-7B-rl-eagle.sh)

File: slime/backends/megatron_utils/megatron_to_hf/qwen3_next.py [hunk 3]
```diff
@@ -1,9 +1,55 @@
...
+        parts = name.split(".")
+        try:
+            layer_idx_loc = parts.index("layers") + 1
+            layer_idx = parts[layer_idx_loc]
+        except (ValueError, IndexError):
+            pass
+        else:
```
Comment (Zhuohao-Li) [inline slime/backends/megatron_utils/megatron_to_hf/qwen3_next.py]: with args fixed, this part could be better if cosnider mtp-layers >1 and ==1 together
